scientific/src/commonTest/kotlin/formatter/CommonScientificValueFormatterTest.kt

```diff
@@ -0,0 +1,111 @@
+import com.splendo.kaluga.scientific.unit.Liter
+import com.splendo.kaluga.scientific.unit.Lumen
+import com.splendo.kaluga.scientific.unit.Lux
+import com.splendo.kaluga.scientific.unit.Megameter
+import com.splendo.kaluga.scientific.unit.Meter
+import com.splendo.kaluga.scientific.unit.Microfarad
+import com.splendo.kaluga.scientific.unit.Micrometer
+import com.splendo.kaluga.scientific.unit.Mile
+import com.splendo.kaluga.scientific.unit.Milliliter
+import com.splendo.kaluga.scientific.unit.Millimeter
+import com.splendo.kaluga.scientific.unit.Nanometer
+import com.splendo.kaluga.scientific.unit.Newton
+import com.splendo.kaluga.scientific.unit.per
+import kotlin.random.Random
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+class CommonScientificValueFormatterTest {
+    lateinit var formatter: ScientificValueFormatter
+
+    @Test
+    fun format__it_converts_scientific_value_to_string() {
+        formatter = CommonScientificValueFormatter
+        val value = randomScientificValue()
+
+        assertIs<String>(formatter.format(value), "It should convert scientific value to string")
+    }
+
+    @Test
+    fun format__defaultFormatter__it_uses_value_and_symbol() {
+        formatter = CommonScientificValueFormatter
+
+        assertEquals("1 m", formatter.format(1(Meter)))
+        assertEquals("2 nm", formatter.format(2(Nanometer)))
+        assertEquals("3 μm", formatter.format(3(Micrometer)))
+        assertEquals("4 mm", formatter.format(4(Millimeter)))
+        assertEquals("5 cm", formatter.format(5(Centimeter)))
+        assertEquals("6 dm", formatter.format(6(Decimeter)))
+        assertEquals("7 dam", formatter.format(7(Decameter)))
+        assertEquals("8 hm", formatter.format(8(Hectometer)))
+        assertEquals("9 km", formatter.format(9(Kilometer)))
+        assertEquals("10 Mm", formatter.format(10(Megameter)))
+        assertEquals("11 Gm", formatter.format(11(Gigameter)))
+
+        assertEquals("1.5 m", formatter.format(1.5(Meter)))
+        assertEquals("10 l", formatter.format(10(Liter)))
+        assertEquals("10 ml", formatter.format(10(Milliliter)))
+        assertEquals("1500 kN", formatter.format(1500(Kilonewton)))
+        assertEquals("65 hP", formatter.format(65(Hectopascal)))
+        assertEquals("0.5 μF", formatter.format(0.5(Microfarad)))
+        assertEquals("16 mi", formatter.format(16(Mile)))
+        assertEquals("1 cd", formatter.format(1(Candela)))
+        assertEquals("11 lm", formatter.format(11(Lumen)))
+        assertEquals("15 lx", formatter.format(15(Lux)))
+        assertEquals("0.342 fc", formatter.format(0.342(FootCandle)))
+        assertEquals("30 km/h", formatter.format(30(Kilometer per Hour)))
+    }
+
+    @Test
+    fun format__custom_format_added__it_uses_custom_formatter() {
+        formatter = CommonScientificValueFormatter.Builder.build {
+            ifUnitIs(Kilometer per Hour) { "$it км/ч" }
+            ifUnitIs(Newton) { "🍏_$it" }
```

Comment:
Some possible alternative syntax options based on discussion with @corrado4eyes : b41080e17a03926e78e998ddaad34c885d91a4ae or 0bf9bac6eb3e5ac096869b9268e5857ada65548c

Reply:
One more option:
```
infix fun ScientificUnit<*>.use(format: CustomFormatHandler) { customFormatters[this] = format }
```
Then
```
(Kilogram per Meter) use { "$it xxx" }
```
This `use` available only inside builder context.

Reply:
Looks like something that @corrado4eyes wanted 🙂

Reply:
`ifUnitIs` doesn't say anything to me. You can keep original syntax, but name should be changed to something clear: like `useFormat`, `customFormatFor`, etc

Reply:
Renamed to `useFormat`

Reply:
👍